src/oversampling_methods/borderline_smote1/interface_mlj.jl

```diff
@@ -0,0 +1,169 @@
+"""
+function BorderlineSMOTE1(;
+    m::Integer = 5,
+    k::Integer = 5,
+    ratios::Union{Nothing,AbstractFloat,Dict{T,<:AbstractFloat}} = 1.0,
+    rng::Union{Integer,AbstractRNG} = default_rng(), try_perserve_type::Bool=true
+) where {T}
+    model = BorderlineSMOTE1(m, k, ratios, rng, try_perserve_type)
+    MMI.clean!(model)
+    return model
+end
+
+"""
+Oversample data X, y using BorderlineSMOTE1
+"""
+function MMI.transform(s::BorderlineSMOTE1, _, X, y)
+    borderline_smote1(X, y; m = s.m, k = s.k, ratios = s.ratios, rng = s.rng, 
+        try_perserve_type = s.try_perserve_type)
+end
+function MMI.transform(s::BorderlineSMOTE1, _, X::AbstractMatrix{<:Real}, y)
+    borderline_smote1(X, y; m = s.m, k = s.k, ratios = s.ratios, rng = s.rng)
+end
+
+
+MMI.metadata_pkg(
+    BorderlineSMOTE1,
+    name = "Imbalance",
+    package_uuid = "c709b415-507b-45b7-9a3d-1767c89fde68",
+    package_url = "https://github.com/JuliaAI/Imbalance.jl",
+    is_pure_julia = true,
+)
+
+MMI.metadata_model(
+    BorderlineSMOTE1,
+    input_scitype = Union{Table(Continuous),AbstractMatrix{Continuous}},
+    output_scitype = Union{Table(Continuous),AbstractMatrix{Continuous}},
+    target_scitype = AbstractVector,
+    load_path = "Imbalance.MLJ.BorderlineSMOTE1"
+)
+function MMI.transform_scitype(s::BorderlineSMOTE1)
+    return Tuple{
+        Union{Table(Continuous),AbstractMatrix{Continuous}},
+        AbstractVector{<:Finite},
+    }
+end
+
+
+
+"""
+$(MMI.doc_header(BorderlineSMOTE1))
+
+`BorderlineSMOTE1` implements the BorderlineSMOTE1 algorithm to correct for class imbalance as in
+Han, H., Wang, W.-Y., & Mao, B.-H. (2005). Borderline-SMOTE: A new over-sampling method in imbalanced data sets learning. 
+In D.S. Huang, X.-P. Zhang, & G.-B. Huang (Eds.), Advances in Intelligent Computing (pp. 878-887). Springer. 
+
+
+# Training data
+
+In MLJ or MLJBase, wrap the model in a machine by
+
+    mach = machine(model)
+
+There is no need to provide any data here because the model is a static transformer.
+
+Likewise, there is no need to `fit!(mach)`.
+
+For default values of the hyper-parameters, model can be constructed by
+
+    model = BorderlineSMOTE1()
+
+
+# Hyperparameters
+
+- `m::Integer=5`: The number of neighbors to consider while checking the BorderlineSMOTE1 condition. In this, a point may participate
+    in oversampling iff the number of neighbors that belong to its class is in \$(0, m/2]\$. Should be within the range 
+   `0 < m < N` where N is the number of observations in the data. It will be automatically set to `N-1` if `N ≤ m`.
+
+- `k::Integer=5`: Number of nearest neighbors to consider in the SMOTE part of the algorithm. Should be within the range
+    `0 < k < n` where n is the number of observations in the smallest class. It will be automatically set to
+    `n-1` for any class where `n ≤ k`.
+
+$((COMMON_DOCS["RATIOS"]))
+
+$((COMMON_DOCS["RNG"]))
+
+# Transform Inputs
+
+$((COMMON_DOCS["INPUTS"]))
+
```

Comment:
This interpolated doc also suggests only continuous input features are allowed. Is this really true?

Reply:
Yes, it is; I justified in a response to another comment. `RandomOversampling`, `RandomUndersampling`, `RandomWalkOversampling`, `SMOTE-NC` all accept continuous and categorical data. `SMOTE-N` accepts only categorical data.